src/EthereumHelpers.js

```diff
@@ -91,9 +91,64 @@ function bytesToRaw(bytesString) {
   return bytesString.replace("0x", "").slice(2)
 }
 
+/**
+ * Takes a bound web3 contract method (e.g. `myContract.methods.doSomething(param)`
+ * and sends it with proper gas estimation and error handling. In particular,
+ * runs a gas estimate beforehand to attach an appropriate gas spend, and, if
+ * the gas estimation fails due to an always-failing transaction, `call`s the
+ * method to get the proper underlying error message. Otherwise, sends the
+ * signed transaction normally.
+ *
+ * @param {*} boundContractMethod A bound web3 contract method with
+ *        `estimateGas`, `send`, and `call` variants available.
+ * @param {*} sendParams The parameters to pass to `estimateGas` and `send` for
+ *        transaction processing.
+ * @param {boolean} forceSend Force the transaction send through even if gas
+ *        estimation fails.
+ *
+ * @return {Promise<any>} A promise to the result of sending the bound contract
+ *         method. Fails the promise if gas estimation fails, extracting an
+ *         on-chain error if possible.
+ */
+async function sendSafely(boundContractMethod, sendParams, forceSend) {
```

Comment:
`forceSend` feels like a workaround for an issue with gas estimation. I haven't encountered it yet, it would be good to document it a bit more here / in an issue.

Reply:
The error handling sounds like it could be solved by setting [`handleRevert`](https://web3js.readthedocs.io/en/v1.2.6/web3-eth-contract.html#handlerevert) in `initializeContracts`, which just came to light. However, happy to consider it nonblocking if there are higher priority things for now.

Reply:
Yes please, definitely let's not block, though I'll likely introduce a PR this morning to add that back.

Agree re: gas estimation. Still don't know what's up there, and perhaps `handleRevert` will help.